packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProvenanceSearchService.java

```diff
@@ -266,6 +266,38 @@ public Set<String> modelsFromEquation(final ProvenanceQueryParam payload) {
 		}
 	}
 
+	/**
+	 *
+	 * Identifies the document from which a model configuration was extracted
+	 *
+	 * @param payload - Search param payload.
+	 * @return
+	 */
+	public Set<String> modelConfigFromDocument(final ProvenanceQueryParam payload) {
+		if (payload.getRootType() != ProvenanceType.MODEL_CONFIGURATION) {
+			throw new IllegalArgumentException(
+					"Document used for model configuration extraction can only be found by providing a Model confirguration");
+		}
+
+		try (final Session session = neo4jService.getSession()) {
+			final UUID modelId = payload.getRootId();
+
+			final String query = String.format("MATCH (d:Document)<-[r:EXTRACTED_FROM]-(m:ModelConfiguration {id: '%s'}) RETURN d",
+					modelId);
+
+			final Result response = session.run(query);
+
+			log.info("Response: " + response.toString());
```

Comment:
Will this and the next log statements get noisy on the back end or are these just for testing?

Reply:
ah these were for testing, Ill remove them